marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/4/triggers/entity-model-trigger.xqy

```diff
@@ -132,8 +132,15 @@ return (
       }, map:entry("database", xdmp:schema-database())
     ),
 
+  (: Check if TDE generation is disabled. Any value other than false or "false" disables TDE generation. :)
```

Comment:
For readability, let's toss this into a function in the module - e.g. local:is-tde-generation-enabled($definitions, $entity-title) . 

Better yet - let's put that function into hub-entities.xqy. You can write in-memory unit tests against that function, which makes it easier to test the scenario where there's not an entity definition whose name matches that of info/title (I don't think that's covered below?). 

It's good to have a test against the trigger too, but I'd opt for one trivial test there, and then exhaustive tests against this function, as the latter will be faster and easier to understand.

Reply:
@rjrudin ,
So, is this response correct for the following 2 entities
hent.isTdeGenerationDisabled(entity1) - false
hent.isTdeGenerationDisabled(entity2) - true
```
  let entity1 = {
    "info": {
      "title": "Person",
      "version": "0.0.1",
      "baseUri": "http://marklogic.com/",
      "description": "A person entity"
    },
    "definitions": {
      "Customer": {
        "tdeGenerationDisabled": "true"
      }
    }
  }
  let entity2 = {
    "info": {
      "title": "Person",
      "version": "0.0.1",
      "baseUri": "http://marklogic.com/",
      "description": "A person entity"
    },
    "definitions": {
      "Person": {
        "tdeGenerationDisabled": "randomValue"
      }
    }
  }
```

Reply:
First, call the function "isTdeGenerationEnabled", as the typical use will be "if TDE generation is enabled, then do the following" (as opposed to "If TDE generation is disabled, then do nothing" or "If TDE generation is not disabled, then do the following", which is a double negative). 

The property should still be tdeGenerationDisabled, since for backwards compatibility reasons, we need to default to enabling it if the property is not there.

For entity1, TDE generation would not be enabled (because there's no entity with a name matching info/title, so we use the first one) and for entity2 it would be enabled because the value is not the boolean value true. (Let's only support the boolean true and not "true" - the user expects to put a boolean here, not a string).

Reply:
Effectively, the rule is - "If the definition whose name matches info/title, or the first definition in the absence of such a match, has tdeGenerationDisabled = true, then do not generate a TDE".

Reply:
Just to confirm, for entity1, TDE generation should be enabled right ? There's no entity with a name matching info/title, so we use the first one("Customer"), it has "tdeGenerationDisabled" set to "true" (which we don't support) and not true.

Reply:
Oh right - yeah, it should be boolean true not "true", so yes, it would be enabled. 

All good explanations to put into assertion messages. :)